release(docker): builder image arm64 hits 30m hard timeout

## Context

The Publish Docker Images workflow currently has two jobs that timeout-fail at the 30-minute hard cap, while the runtime image jobs succeed in under 1 minute.

From run 24019393053 on master (workflow_dispatch, 2026-04-05):

\`\`\`
✓ Build perl-lsp runtime image (linux/amd64) in 56s
✓ Build perl-lsp runtime image (linux/arm64) in 51s
✓ Publish perl-lsp runtime to Docker Hub in 35m50s
✓ Build builder image (linux/amd64) in 9m13s
X Build builder image (linux/arm64) in 30m8s   ← timeout
X Publish to Docker Hub                in 30m17s ← timeout
##[error] The job has exceeded the maximum execution time of 30m0s
\`\`\`

The runtime image (the actual user-facing release artifact) is fine and shipping — \`effortlessmetrics/perl-lsp:0.12.2-perl\` last_updated 2026-04-06 confirms it. The failing jobs are the heavier "builder image" variant on linux/arm64 and the consolidated Docker Hub publish job, which both depend on a slow QEMU arm64 emulation path.

## Why this matters

These timeouts make the Docker publish workflow appear red on the Actions page even though the user-facing image shipped successfully. During the v0.12.2 release verification this caused real confusion ("Docker is broken — release isn't done") when in fact only the developer-facing builder image was affected.

## Fix options

1. **Bump \`timeout-minutes\` to 60–90** for the arm64 builder image and consolidated Docker Hub publish jobs — cheapest, just stops the false red.
2. **Cross-compile builder image instead of QEMU emulation** — meaningfully faster and more correct, but a larger change.
3. **Make builder image arm64 a non-blocking matrix entry** (\`continue-on-error: true\`) — preserves visibility without poisoning the run status.

Option 1 is the smallest fix; option 2 is the right long-term answer.

## Acceptance criteria

- Publish Docker Images workflow runs cleanly green for the next release without manual reruns or cancellations.
- \`effortlessmetrics/perl-lsp:builder-arm64\` (or whatever the builder image's tag is) publishes alongside the runtime image.
- Workflow file: \`.github/workflows/docker-publish.yml\`.

--- a/.docker/rust/Dockerfile
+++ b/.docker/rust/Dockerfile
@@ -1,4 +1,4 @@
-FROM rust:slim
+FROM rust:1.92-slim-bookworm
 
 # Install system dependencies
 RUN apt-get update && apt-get install -y --no-install-recommends \

--- a/.github/workflows/docker-publish.yml
+++ b/.github/workflows/docker-publish.yml
@@ -52,7 +52,7 @@ jobs:
     name: Build builder image (${{ matrix.platform }})
     runs-on: ubuntu-24.04
     needs: init
-    timeout-minutes: 30
+    timeout-minutes: 90
     strategy:
       fail-fast: false
       matrix:
@@ -162,7 +162,7 @@ jobs:
     name: Publish to Docker Hub
     runs-on: ubuntu-24.04
     needs: init
-    timeout-minutes: 30
+    timeout-minutes: 90
     if: github.event.inputs.push != 'false'
 
     steps:

